de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/SarosSessionManager.java

```diff
@@ -561,37 +616,34 @@ public void addResourcesToSession(
         }
 
         INegotiationHandler handler = negotiationHandler;
-
         if (handler == null) {
             log.warn("could not start a project negotiation because no handler is installed");
             return;
         }
 
         List<AbstractOutgoingProjectNegotiation> negotiations = new ArrayList<AbstractOutgoingProjectNegotiation>();
 
-        synchronized (this) {
-            if (!startStopSessionLock.tryLock()) {
-                log.warn("could not start a project negotiation because the current session is about to stop");
-                return;
-            }
+        if (!startStopSessionLock.tryLock()) {
+            log.warn("could not start a project negotiation because the current session is about to stop");
+            return;
+        }
 
-            try {
-                for (User user : currentSession.getRemoteUsers()) {
-
-                    TransferType type = TransferType.valueOf(currentSession
-                        .getUserProperties(user).getString(
-                            ProjectNegotiationTypeHook.KEY_TYPE));
-                    AbstractOutgoingProjectNegotiation negotiation = negotiationFactory
-                        .newOutgoingProjectNegotiation(user.getJID(), type,
-                            projectsToShare, this, currentSession);
-
-                    negotiation.setNegotiationListener(negotiationListener);
-                    currentProjectNegotiations.add(negotiation);
-                    negotiations.add(negotiation);
-                }
-            } finally {
-                startStopSessionLock.unlock();
+        try {
+            for (User user : currentSession.getRemoteUsers()) {
+
+                TransferType type = TransferType.valueOf(currentSession
+                    .getUserProperties(user).getString(
+                        ProjectNegotiationTypeHook.KEY_TYPE));
+                AbstractOutgoingProjectNegotiation negotiation = negotiationFactory
+                    .newOutgoingProjectNegotiation(user.getJID(), type,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "getJID"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=272&issues=AWb814lgTJjFgJLT72lr&open=AWb814lgTJjFgJLT72lr)

[ISSUE_KEY]: AWb814lgTJjFgJLT72lr